lsp-haskell.el

```diff
@@ -228,6 +228,25 @@ if projectile way fails"
           (user-error "Couldn't find cabal file, using: %s" dir)
         dir))))
 
+
+;; ---------------------------------------------------------------------
+;; Interactive commands
+
+(defun lsp-haskell-case-split ()
+  "Case split on an identifier.
+To use this, place make sure the point is over a type hole."
+  (interactive)
+  (let* ((case-split-regex "Case split on \\(.*\\)")
+         (actions (-filter (lambda (action) (s-matches-p case-split-regex (lsp:code-action-title action))) (lsp-code-actions-at-point))))
+    (lsp-execute-code-action (cond ((seq-empty-p actions) (signal 'lsp-no-code-actions nil))
+                                   ((and (eq (seq-length actions) 1) lsp-auto-execute-action)
+                                    (lsp-seq-first actions))
+                                   (t (lsp--completing-read "Identifier: "
+                                                            (seq-into actions 'list)
+                                                            (lambda (action)
+                                                              (cadr (s-match case-split-regex (lsp:code-action-title action))))
```

Comment:
Hmm, you do rather rely on the regex here. I guess I don't have a better suggestion, then, since AFAICT there's no proper way to include some extra structured data in a `CodeAction`...

I guess we could avoid the regex by just using the code action title as is, for something like:
```
Pick a case-splitting action:
1. Case split on x
2. Case split on y
```
but I guess that's a worse UX.

Reply:
If it is too fragile, then we can remove it! Shame that the spec doesn't let us communicate more in the kind.

Reply:
Yeah, I just want to avoid the situation where someone changes the title, rightly reasoning that it's a user-facing string that can be changed freely, and then this function breaks.